ingestion-sink/src/test/java/com/mozilla/telemetry/ingestion/sink/util/TestWithPubsubResources.java

```diff
@@ -14,32 +15,45 @@
 import com.google.cloud.pubsub.v1.SubscriptionAdminSettings;
 import com.google.cloud.pubsub.v1.TopicAdminClient;
 import com.google.cloud.pubsub.v1.TopicAdminSettings;
-import com.google.pubsub.v1.ProjectSubscriptionName;
-import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
 import com.google.pubsub.v1.PushConfig;
+import com.google.pubsub.v1.Subscription;
 import io.grpc.ManagedChannelBuilder;
 import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
 import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import org.junit.After;
 import org.junit.Before;
 
 public abstract class TestWithPubsubResources {
 
-  public String projectId;
-  public String topicId;
-  public String subscriptionId;
-  public ProjectTopicName topicName;
-  public ProjectSubscriptionName subscriptionName;
-  public TopicAdminClient topicAdminClient;
-  public SubscriptionAdminClient subscriptionAdminClient;
+  private String projectId;
+  protected int numTopics = 1;
```

Comment:
Maybe we should make `numTopics` abstract so that clients are forced to explicitly set it. That would perhaps make it easier to understand why we're indexing methods with `0` in various places.

Reply:
Or maybe we should have a `TestWithSinglePubsubTopic` subclass that sets numTopics = 1 and provides simpler methods that avoid having to pass a `0` index.

Reply:
I like this option, will do